libraries/AP_Torqeedo/AP_Torqeedo.cpp

```diff
@@ -289,6 +383,55 @@ void AP_Torqeedo::send_motor_speed_cmd()
 
     _last_send_motor_us = AP_HAL::micros();
     _send_delay_us = calc_send_delay_us(buff_size);
+
+    {
+        // record time of send for health reporting
+        WITH_SEMAPHORE(_last_healthy_sem);
+        _last_send_motor_ms = AP_HAL::millis();
+    }
+
+}
+
+// output logging and debug messages (if required)
+void AP_Torqeedo::log_and_debug()
+{
+    // exit immediately if debugging is disabled
+    if (_debug_level == DebugLevel::NONE) {
+        return;
+    }
+
+    // return if not enough time has passed since last output
+    const uint32_t now_ms = AP_HAL::millis();
+    if (now_ms - _last_debug_ms < (1000 * TORQEEDO_LOG_INTERVAL)) {
+        return;
+    }
+
+    const bool health = healthy();
+    const int16_t mot_speed = health ? _motor_speed : 0;
+
+    // @LoggerMessage: TRQD
+    // @Description: Torqeedo Status
+    // @Field: TimeUS: Time since system startup
+    // @Field: Health: Health
+    // @Field: MotSpeed: Motor Speed (-1000 to 1000)
+    // @Field: SuccCnt: Success Count
+    // @Field: ErrCnt: Error Count
+    AP::logger().Write("TRQD", "TimeUS,Health,MotSpeed,SuccCnt,ErrCnt", "QbhII",
+                       AP_HAL::micros64(),
+                       (int)health,
+                       (int)mot_speed,
+                       (unsigned long)_parse_success_count,
+                       (unsigned long)_parse_error_count);
```

Comment:
Looks like this might have been converted from a printf....

```suggestion
    AP::logger().Write("TRQD", "TimeUS,Health,MotSpeed,SuccCnt,ErrCnt", "QBHII",
                       AP_HAL::micros64(),
                       health,
                       mot_speed,
                       _parse_success_count,
                       _parse_error_count);
```

Reply:
fixed logging, txs!